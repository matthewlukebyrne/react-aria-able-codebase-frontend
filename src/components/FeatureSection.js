// import logo for now
import landing_image from '../img/landing_image.svg';

// import icons from Font Awesome
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCoffee } from '@fortawesome/free-solid-svg-icons';

// Styled Components
import styled from 'styled-components';
import { StyledLayout, StyledDescription, StyledImage } from '../styles';

const FeatureSection = () => {
  return (
    <StyledFeatures data-testid='featuretest-id'>
      {/* Side Main Features Image */}
      <StyledImage>
        <img src={landing_image} alt='features_image' />
      </StyledImage>
      <StyledDescription>
        {/* Title */}
        <h2>
          High <span>quality</span> components
        </h2>
        {/* 4 Cards Replicated */}
        <div className='cards'>
          <div className='card'>
            <div className='icon'>
              <FontAwesomeIcon icon={faCoffee} title='icon-1' />
              <h3>Lorem ipsum dolor sit amet.</h3>
            </div>
            <p>Lorem ipsum dolor sit amet.</p>
          </div>
          <div className='card'>
            <div className='icon'>
              <FontAwesomeIcon icon={faCoffee} title='icon-2' />
              <h3>Lorem ipsum dolor sit amet.</h3>
            </div>
            <p>Lorem ipsum dolor sit amet.</p>
          </div>
          <div className='card'>
            <div className='icon'>
              <FontAwesomeIcon icon={faCoffee} title='icon-3' />
              <h3>Lorem ipsum dolor sit amet.</h3>
            </div>
            <p>Lorem ipsum dolor sit amet.</p>
          </div>
          <div className='card'>
            <div className='icon'>
              <FontAwesomeIcon icon={faCoffee} title='icon-4' />
              <h3>Lorem ipsum dolor sit amet.</h3>
            </div>
            <p>Lorem ipsum dolor sit amet.</p>
          </div>
        </div>
      </StyledDescription>
    </StyledFeatures>
  );
};

export default FeatureSection;

// Styled Components Area
const StyledFeatures = styled(StyledLayout)`
  background-color: whitesmoke;
  color: #333;

  h2,
  p {
    padding: 5rem 0rem;
    color: #333;
  }

  img {
    width: 100%;
    height: 70vh;
    object-fit: cover;
  }

  // Inserted some mobile media queries (drop it when it shrinks)
  @media (max-width: 1300px) {
    display: none;
  }
`;

// const StyledCards = styled.div`
// 	margin: 5rem 0rem;
// 	display: flex;
// 	flex-wrap: wrap;
// 	font-size: 1.5rem;
// `;

// const StyledCard = styled.div`
// 	flex-basis: 100rem;
// 	h3 {
// 		margin-left: 5rem;
// 		background: #fff;
// 		border-radius: 16px;
// 		color: #1b1b1b;
// 		padding: 1rem;
// 	}
//  Font Awesome Icons
// .icon {
//   display: flex;
//   align-items: center;
// }
// `;
