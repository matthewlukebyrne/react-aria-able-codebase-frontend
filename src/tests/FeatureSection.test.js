import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import FeatureSection from '../components/FeatureSection';
import renderer from 'react-test-renderer';


// TDD - Red Green Refactor
describe('Feature Section Component Unit Tests', () => {
  // the first test
  it('render the feature section component without crashing', () => {
    render(<FeatureSection />);
    const featureSectionPonent = screen.getByTestId('featuretest-id');
    // make assertions
    expect(featureSectionPonent).toBeInTheDocument();
  });

  // the first test
  it('should render the four Font Awesome Icons on the page', () => {
    render(<FeatureSection />);
    const icon_1 = screen.getByTitle('icon-1');
    const icon_2 = screen.getByTitle('icon-2');
    const icon_3 = screen.getByTitle('icon-3');
    const icon_4 = screen.getByTitle('icon-4');
    // make assertions
    expect(icon_1).toBeInTheDocument();
    expect(icon_2).toBeInTheDocument();
    expect(icon_3).toBeInTheDocument();
    expect(icon_4).toBeInTheDocument();
  });

  it('feature section should contain four H3 tags', () => {
    const featureQuery = shallow(<FeatureSection />);
    const featureH3Tags = featureQuery.find('h3');
    expect(featureH3Tags).toHaveLength(4);
  });

  test('feature section should render in whitesmoke', () => {
    const landingStyles = renderer.create(<FeatureSection />).toJSON();
    expect(landingStyles).toMatchSnapshot();
    expect(landingStyles).toHaveStyleRule('background-color', 'whitesmoke');
  });
});
