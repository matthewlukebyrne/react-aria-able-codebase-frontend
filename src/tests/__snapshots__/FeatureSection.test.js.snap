// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Section Component Unit Tests feature section should render in whitesmoke 1`] = `
.c0 {
  min-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0rem 10rem;
  color: white;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-right: 5rem;
  z-index: 2;
}

.c3 h2 {
  padding-bottom: 1rem;
  font-weight: lighter;
}

.c2 {
  z-index: 2;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c2 img {
  padding: 2rem;
  width: 100%;
  height: 80vh;
  object-fit: cover;
}

.c1 {
  background-color: whitesmoke;
  color: #333;
}

.c1 h2,
.c1 p {
  padding: 5rem 0rem;
  color: #333;
}

.c1 img {
  width: 100%;
  height: 70vh;
  object-fit: cover;
}

@media (max-width:1300px) {
  .c0 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c3 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c2 {
    display: block;
    padding: 5rem 2rem;
    text-align: center;
  }

  .c2 img {
    width: 100%;
    height: 50vh;
    object-fit: cover;
  }
}

@media (max-width:1300px) {
  .c1 {
    display: none;
  }
}

<div
  className="c0 c1"
  data-testid="featuretest-id"
>
  <div
    className="c2"
  >
    <img
      alt="features_image"
      src="landing_image.svg"
    />
  </div>
  <div
    className="c3"
  >
    <h2>
      High 
      <span>
        quality
      </span>
       components
    </h2>
    <div
      className="cards"
    >
      <div
        className="card"
      >
        <div
          className="icon"
        >
          <svg
            aria-labelledby="svg-inline--fa-title-nR6CjKFqNvC2"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="svg-inline--fa-title-nR6CjKFqNvC2"
              style={Object {}}
            >
              icon-1
            </title>
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h3>
            Lorem ipsum dolor sit amet.
          </h3>
        </div>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </div>
      <div
        className="card"
      >
        <div
          className="icon"
        >
          <svg
            aria-labelledby="svg-inline--fa-title-qgzs0JoJDcKQ"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="svg-inline--fa-title-qgzs0JoJDcKQ"
              style={Object {}}
            >
              icon-2
            </title>
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h3>
            Lorem ipsum dolor sit amet.
          </h3>
        </div>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </div>
      <div
        className="card"
      >
        <div
          className="icon"
        >
          <svg
            aria-labelledby="svg-inline--fa-title-8DdiN68LqEo7"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="svg-inline--fa-title-8DdiN68LqEo7"
              style={Object {}}
            >
              icon-3
            </title>
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h3>
            Lorem ipsum dolor sit amet.
          </h3>
        </div>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </div>
      <div
        className="card"
      >
        <div
          className="icon"
        >
          <svg
            aria-labelledby="svg-inline--fa-title-mmyrs5CwFDDh"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="svg-inline--fa-title-mmyrs5CwFDDh"
              style={Object {}}
            >
              icon-4
            </title>
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <h3>
            Lorem ipsum dolor sit amet.
          </h3>
        </div>
        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </div>
    </div>
  </div>
</div>
`;
